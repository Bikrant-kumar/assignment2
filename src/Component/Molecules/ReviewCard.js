import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import Label from '../Atoms/Label';
import Rating from '../Atoms/Rating';

const useStyles = makeStyles((theme) => ({
    root: {
      display: 'grid',
      gridTemplateRow: 'repeat(12, 1fr)',
      gridGap: theme.spacing(1),
      marginTop: theme.spacing(2),
      width: "43%",
      marginLeft: "5%",
      borderRadius: "10px",
    },
    divider: {
      margin: theme.spacing(0.1),
    },
  }));

export default function ReviewCard(props) {

  const classes = useStyles();
    debugger
  return (
    <Card className={classes.root}>
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2" >
            <Label text={props.review.title}/>
          </Typography>  
        <Typography variant="body2" color="textSecondary" component="p">
             {props.review.comments}  
        </Typography>
        </CardContent>
        <Divider className={classes.divider} />
        <Typography variant="body2" component="p">
            <Rating defaultValue={props.review.rating} readOnly="false"/>  
        </Typography>
    </Card>
  );
}