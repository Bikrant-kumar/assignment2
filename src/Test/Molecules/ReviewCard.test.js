import React from 'react';
import { screen, render } from '@testing-library/react';
import ReviewCard from '../../Component/Molecules/ReviewCard';
import renderer from 'react-test-renderer';

describe('ReviewCard',()=> {

    test('ReviewCard Render',() => {
  
      const review={
          "id": "review-1",
          "title": "Biryani",
          "comments":"Quality and quantity wise its a perfect biryani brand",
          "rating": "4.5"
      };      
          render(<ReviewCard review={review}/>);
          expect(screen.getByText("Biryani")).toBeInTheDocument();
       });
  
       test('test ReviewCard with snapshot',()=>{
        const review={
          "id": "review-1",
          "title": "Biryani",
          "comments":"Quality and quantity wise its a perfect biryani brand",
          "rating": "4.5"
      };
        const component= renderer.create(<ReviewCard review = {review}/>).toJSON();
        expect(component).toMatchSnapshot();
       });
  
  })
   