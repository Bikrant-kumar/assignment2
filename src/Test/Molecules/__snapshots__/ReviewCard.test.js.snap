// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewCard test ReviewCard with snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-5 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      <div
        className="MuiBox-root MuiBox-root-7"
      >
        Biryani
      </div>
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      Quality and quantity wise its a perfect biryani brand
    </p>
  </div>
  <hr
    className="MuiDivider-root makeStyles-divider-6"
  />
  <p
    className="MuiTypography-root MuiTypography-body2"
  >
    <div
      className="makeStyles-root-8"
    >
      <span
        aria-label="4.5 Stars"
        className="MuiRating-root MuiRating-sizeLarge MuiRating-readOnly"
        data-testid="checkRating"
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        role="img"
      >
        <span
          className="MuiRating-decimal"
        >
          <span
            style={
              Object {
                "overflow": "hidden",
                "position": "absolute",
                "width": "0%",
                "zIndex": 1,
              }
            }
          >
            <span
              className="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style={Object {}}
          >
            <span
              className="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          className="MuiRating-decimal"
        >
          <span
            style={
              Object {
                "overflow": "hidden",
                "position": "absolute",
                "width": "0%",
                "zIndex": 1,
              }
            }
          >
            <span
              className="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style={Object {}}
          >
            <span
              className="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          className="MuiRating-decimal"
        >
          <span
            style={
              Object {
                "overflow": "hidden",
                "position": "absolute",
                "width": "0%",
                "zIndex": 1,
              }
            }
          >
            <span
              className="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style={Object {}}
          >
            <span
              className="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          className="MuiRating-decimal"
        >
          <span
            style={
              Object {
                "overflow": "hidden",
                "position": "absolute",
                "width": "0%",
                "zIndex": 1,
              }
            }
          >
            <span
              className="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style={Object {}}
          >
            <span
              className="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          className="MuiRating-decimal"
        >
          <span
            style={
              Object {
                "overflow": "hidden",
                "position": "absolute",
                "width": "50%",
                "zIndex": 1,
              }
            }
          >
            <span
              className="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span
            style={Object {}}
          >
            <span
              className="MuiRating-icon MuiRating-iconEmpty"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
      </span>
    </div>
  </p>
</div>
`;
